VirtoCommerce.CoreModule.Data/Services/CommerceServiceImpl.cs

```diff
@@ -188,6 +188,32 @@ public void DeleteSeoForObject(coreModel.ISeoSupport seoSupportObject)
             }
         }
 
+        public void DeleteSeoForObjects(coreModel.ISeoSupport[] seoSupportObjects)
+        {
+            if (seoSupportObjects == null)
+            {
+                throw new ArgumentNullException(nameof(seoSupportObjects));
+            }
+
+            foreach (var seoObject in seoSupportObjects.Where(x => x.Id != null))
+            {
+                using (var repository = _repositoryFactory())
+                {
+                    var objectType = seoObject.SeoObjectType;
+                    var objectId = seoObject.Id;
+                    var seoUrlKeywords = repository.GetObjectSeoUrlKeywords(objectType, objectId);
+
+                    foreach (var seoUrlKeyword in seoUrlKeywords)
+                    {
+                        repository.Remove(seoUrlKeyword);
+                    }
+
+                    CommitChanges(repository);
```

Comment:
Could we place CommitChanges after the cycle to not send the query after each object keywords removing?